framework/include/base/MooseApp.h

```diff
@@ -350,6 +351,12 @@ class MooseApp : public ConsoleStreamInterface,
                          const std::string & name,
                          const InputParameters & params);
 
+  /**
+   * Helper function to link new added Builder back to Parser. This function will be removed after
+   *new Parser and builder are merged
+   **/
+  Builder & parser() { return _builder; }
```

Comment:
Can we move this to the source file and add a `mooseDeprecated()` message? Something like

> MooseApp::parser() is deprecated, use MooseApp::builder() instead.